adapters/cisco_ios_xr/cisco_ios_xr_apply_conf.php

```diff
@@ -34,169 +34,139 @@ function cisco_ios_xr_apply_conf($configuration, $push_to_startup = false)
   $ret = SMS_OK;
-  // if ($line_config_mode === 1)
-  {
-    echo "Line by line mode configuration\n";
-    $ERROR_BUFFER = '';
 
-    sendexpectone(__FILE__ . ':' . __LINE__, $sms_sd_ctx, "conf t", "(config)#", DELAY);
+  echo "Line by line mode configuration\n";
+  $ERROR_BUFFER = '';
 
-    unset($tab);
-    $tab[0] = $sms_sd_ctx->getPrompt();
-    $tab[1] = ")#";
-    $tab[2] = "]?";
-    $tab[3] = "[confirm]";
-    $tab[4] = "[no]:";
+  sendexpectone(__FILE__ . ':' . __LINE__, $sms_sd_ctx, "conf exclusive", "(config)#", DELAY);
 
-    $buffer = $configuration;
-    $line = get_one_line($buffer);
-    while ($line !== false)
+  unset($tab);
+  $tab[0] = $sms_sd_ctx->getPrompt();
+  $tab[1] = ")#";
+  $tab[2] = "]?";
+  $tab[3] = "[confirm]";
+  $tab[4] = "[no]:";
+
+  $buffer = $configuration;
+  $line = get_one_line($buffer);
+  while ($line !== false)
+  {
+    $line = trim($line);
+    if (strpos($line, "!") === 0)
     {
-      $line = trim($line);
-      if (strpos($line, "!") === 0)
+      echo "$sdid: $line\n";
+    }
+    else
+    {
+      $index = sendexpect(__FILE__ . ':' . __LINE__, $sms_sd_ctx, $line, $tab, DELAY);
+      $SMS_OUTPUT_BUF .= $sendexpect_result;
+      if (($index === 2) || ($index === 3))
       {
-        echo "$sdid: $line\n";
+        sendexpect(__FILE__ . ':' . __LINE__, $sms_sd_ctx, "", $tab, DELAY);
+        $SMS_OUTPUT_BUF .= $sendexpect_result;
       }
-      else
+      else if ($index === 4)
       {
-        $index = sendexpect(__FILE__ . ':' . __LINE__, $sms_sd_ctx, $line, $tab, DELAY);
+        sendexpect(__FILE__ . ':' . __LINE__, $sms_sd_ctx, "yes", $tab, DELAY);
         $SMS_OUTPUT_BUF .= $sendexpect_result;
-        if (($index === 2) || ($index === 3))
-        {
-          sendexpect(__FILE__ . ':' . __LINE__, $sms_sd_ctx, "", $tab, DELAY);
-          $SMS_OUTPUT_BUF .= $sendexpect_result;
-        }
-        else if ($index === 4)
-        {
-          sendexpect(__FILE__ . ':' . __LINE__, $sms_sd_ctx, "yes", $tab, DELAY);
-          $SMS_OUTPUT_BUF .= $sendexpect_result;
-        }
+      }
 
-        foreach ($apply_errors as $apply_error)
+      foreach ($apply_errors as $apply_error)
+      {
+        if (preg_match($apply_error, $SMS_OUTPUT_BUF, $matches) > 0)
         {
-          if (preg_match($apply_error, $SMS_OUTPUT_BUF, $matches) > 0)
-          {
-            $ERROR_BUFFER .= "!";
-            $ERROR_BUFFER .= "\n";
-            $ERROR_BUFFER .= $line;
-            $ERROR_BUFFER .= "\n";
-            $ERROR_BUFFER .= $apply_error;
-            $ERROR_BUFFER .= "\n";
-            $SMS_OUTPUT_BUF = '';
-          }
+          $ERROR_BUFFER .= "!";
+          $ERROR_BUFFER .= "\n";
+          $ERROR_BUFFER .= $line;
+          $ERROR_BUFFER .= "\n";
+          $ERROR_BUFFER .= $apply_error;
+          $ERROR_BUFFER .= "\n";
+          $SMS_OUTPUT_BUF = '';
+
+          sms_log_error ( __FILE__ . ':' . __LINE__ . ": [[!!! $SMS_OUTPUT_BUF !!!]]\n" );
```

Comment:
line useless because SMS_OUTPUT_BUF  is empty

Reply:
SMS_OUTPUT_BUF to be used. Fixed in commit ad96493: emptied later.